src/hash_join.rs

```diff
@@ -0,0 +1,37 @@
+// http://rosettacode.org/wiki/Hash_join
+
+use std::hash::{hash, Hash, SipHasher};
+use std::collections::HashMap;
+
+pub fn main() {
+    let table1 = vec![(27, "Jonah"),
+              (18, "Alan"),
+              (28, "Glory"),
+              (18, "Popeye"),
+              (28, "Alan")];
+    let table2 = vec![("Jonah", "Whales"),
+              ("Jonah", "Spiders"),
+              ("Alan", "Ghosts"),
+              ("Alan", "Zombies"),
+              ("Glory", "Buffy")];
+    let mut hash_map: HashMap<&str, Vec<&str>> = HashMap::new();
+    
+    //hash phase
+    for (name, nemesis) in table2 {
+        if hash_map.contains_name(name) {
```

Comment:
You could use the `entry` function here:

```
match hash_map.entry(name) {
    Entry::Occupied(entry) => entry.get_mut().push(nemesis)
    Entry::Vacant(entry) => entry.insert(vec![nemesis])
}
```

*I have not tested this, but it should work. You can take a look [here](http://static.rust-lang.org/doc/master/std/collections/hash_map/enum.Entry.htm) or ask on IRC for more information.